### PR TITLE
chore(backend): Merge v2 code into backend/src/

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: go-licenses.yaml [hunk 1]
```diff
@@ -170,9 +170,9 @@ module:
       path: LICENSE
       # gopkg.in/yaml.v2 is port of a MIT licensed library
       spdxId: Apache-2.0 / MIT
-  - name: go.opencensus.io
-    version: v0.22.5
-    license:
-      path: LICENSE
-      url: https://github.com/census-instrumentation/opencensus-go/blob/v0.22.5/LICENSE
-      spdxId: Apache-2.0
+  # - name: go.opencensus.io
```
Comment (chensun) [inline go-licenses.yaml:173]: @Bobgy can you share some knowledge about this file? Would would we need an override entry as such?

**Comment by chensun** [comment]
> @chensun: The following tests **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> kubeflow-pipelines-v2-go-test	[1421952](https://github.com/kubeflow/pipelines/commit/142195223eec2dbec30ab121b96ade084bcc870c)	[link](https://oss-prow.knative.dev/view/gs/oss-prow/pr-logs/pull/kubeflow_pipelines/7173/kubeflow-pipelines-v2-go-test/1484579734439333888)	true	`/test kubeflow-pipelines-v2-go-test`
> kubeflow-pipelines-samples-v2	[1421952](https://github.com/kubeflow/pipelines/commit/142195223eec2dbec30ab121b96ade084bcc870c)	[link](https://oss-prow.knative.dev/view/gs/oss-prow/pr-logs/pull/kubeflow_pipelines/7173/kubeflow-pipelines-samples-v2/1484579734418362368)	true	`/test kubeflow-pipelines-samples-v2`

These two tests are failing due to code being moved. I plan to ignore the failures first, and then update the test configs immediately after this PR is merged.
/cc @zijianjoy Let me know if you have other thoughts.

**Comment by Bobgy** [comment]
Error for v2 sample test:
```
ERROR: ../../sdk/python is not a valid editable requirement.
```

You need to fix relative path in v2/test/requirements.txt.

**Comment by Bobgy** [comment]
License updates LGTM, some changes are not ideal like
```
github.com/gogo/protobuf, https://github.com/gogo/protobuf/blob/v1.3.2/LICENSE, BSD-3-Clause / BSD-2-Clause
```
is now `BSD-3-Clause` only.

However, I think the result is good enough considering that we don't need to maintain the override config.

**Comment by chensun** [comment]
> @chensun: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> kubeflow-pipelines-samples-v2	[78ce469](https://github.com/kubeflow/pipelines/commit/78ce4692d1a9f958bb2942d430bfbe1ea25adac9)	[link](https://oss-prow.knative.dev/view/gs/oss-prow/pr-logs/pull/kubeflow_pipelines/7173/kubeflow-pipelines-samples-v2/1485896461471191040)	true	`/test kubeflow-pipelines-samples-v2`

The tests are quite flaky with some randomly failed executors failed with message: This step is in Error state with this message: Pod failed before main container starts.
Will merge this change as is and look into the test issues afterwards.

**Pull request merged**